Tools/PowershellModule/src/Library/Get-WinGetManifest.ps1

```diff
@@ -82,64 +73,90 @@ Function Get-WinGetManifest
+                if(!$Result) {
+                    Write-Error "Failed to confirm resources exist in Azure. Please verify and try again." -ErrorAction Stop
+                }
 
-                $AzureResourceGroupName = $(Get-AzFunctionApp).Where({$_.Name -eq $FunctionName}).ResourceGroupName
+                ## Retrieves the Azure Function URL used to add new manifests to the REST source
+                Write-Verbose -Message "Retrieving Azure Function Web Applications matching to: $FunctionName."
+                $FunctionApp = Get-AzWebApp -ResourceGroupName $ResourceGroupName -Name $FunctionName
 
-                if($AzureResourceGroupName) {
-                    $Result = Test-AzureResource -FunctionName $FunctionName -ResourceGroup $AzureResourceGroupName
-                }
-                else {
-                    throw "Unable to locate Function (""$FunctionName"") in the Azure Tenant."
-                }
+                $FunctionAppId   = $FunctionApp.Id
+                $DefaultHostName = $FunctionApp.DefaultHostName
+
+                $TriggerName    = "ManifestGet"
+                $ApiMethod      = "Get"
+
+                ## Creates the API Post Header
+                $ApiHeader = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+                $ApiHeader.Add("Accept", 'application/json')
+                $FunctionKey = (Invoke-AzResourceAction -ResourceId "$FunctionAppId/functions/$TriggerName" -Action listkeys -Force).default
+                $ApiHeader.Add("x-functions-key", $FunctionKey)
+            }
+            "File" {
+                ## Nothing to prepare
+            }
+        }        
+    }
+    PROCESS
+    {
+        switch ($PsCmdlet.ParameterSetName) {
+            "Azure" {
+                $AzFunctionURL = "https://" + $DefaultHostName + "/api/packageManifests/" + $PackageIdentifier
                 
-                if(!$Result) {
-                    throw "Failed to confirm resources exist in Azure. Please verify and try again."
+                ## Publishes the Manifest to the Windows Package Manager REST source
+                Write-Verbose -Message "Invoking the REST API call."
+
+                $Response = Invoke-RestMethod $AzFunctionURL -Headers $ApiHeader -Method $ApiMethod -ErrorVariable ErrorInvoke
+
+                if($ErrorInvoke) {
+                    $ErrorMessage = "Failed to get Manifest from $FunctionName. Verify the information you provided and try again."
+                    $ErrReturnObject = @{
+                        AzFunctionURL       = $AzFunctionURL
+                        ApiMethod           = $ApiMethod
+                        Response            = $Response
+                        InvokeError         = $ErrorInvoke
+                    }
+                
+                    Write-Error -Message $ErrorMessage -TargetObject $ErrReturnObject
```

Comment:
Does this need `-ErrorAction Stop` or a `Return` after this line

Reply:
It does not need -ErrorAction Stop since we want to process next item in the pipeline. -ErrorAction Stop will stop processing the whole pipeline